uprotocol/uri/validator/urivalidator.py

```diff
@@ -199,3 +199,21 @@ def matches(uri_to_match: UUri, candidate_uri: UUri) -> bool:
             and UriValidator.matches_entity(uri_to_match, candidate_uri)
             and UriValidator.matches_resource(uri_to_match, candidate_uri)
         )
+
+    @staticmethod
+    def has_wildcard(uri):
+        """
+        Checks if the URI has a wildcard in any of its fields.
+
+        :param uri: The URI to check for wildcards.
+        :type uri: UUri
+
+        :return: True if the URI has a wildcard, False otherwise.
+        :rtype: bool
+        """
+        return uri and (
+            uri.authority_name == UriFactory.WILDCARD_AUTHORITY
+            or (uri.ue_id & UriFactory.WILDCARD_ENTITY_ID) == UriFactory.WILDCARD_ENTITY_ID
```

Comment:
tried to figure out how this worked myself, got confused. If doing that & makes sense to you, then all good w me.

Reply:
This is correct, it is a bitwise AND operation since 1ffff, 2ffff, and similar values are all wildcard entities.

Reply:
cool